tests/startup_capture_test.go

```diff
@@ -0,0 +1,270 @@
+
+	if err != nil {
+		g := context.getWholeGraph(t, time.Now())
+		helper.ExecCmds(t, test.tearDownCmds...)
+		context.getSystemState(t)
+		t.Fatalf("Failed to setup captures: %s, graph: %s", err.Error(), g)
+	}
+
+	retries := test.retries
+	if retries <= 0 {
+		retries = 30
+	}
+
+	settleTime := time.Now()
+
+	if test.settleFunction != nil {
+		err = common.Retry(func() error {
+			return test.settleFunction(context)
+		}, retries, time.Second)
+
+		if err != nil {
+			g := context.getWholeGraph(t, settleTime)
+			f := context.getAllFlows(t, settleTime)
+			helper.ExecCmds(t, test.tearDownCmds...)
+			context.getSystemState(t)
+			t.Errorf("Test failed to settle: %s, graph: %s, flows: %s", err.Error(), g, f)
+			return
+		}
+	}
+
+	context.setupTime = time.Now()
+
+	if test.setupFunction != nil {
+		if err = test.setupFunction(context); err != nil {
+			g := context.getWholeGraph(t, context.setupTime)
+			f := context.getAllFlows(t, context.setupTime)
+			helper.ExecCmds(t, test.tearDownCmds...)
+			context.getSystemState(t)
+			t.Fatalf("Failed to setup test: %s, graph: %s, flows: %s", err.Error(), g, f)
+		}
+	}
+
+	test.checkContexts = make([]*CheckContext, len(test.checks))
+
+	for i, check := range test.checks {
+		checkContext := &CheckContext{
+			TestContext: context,
+			startTime:   time.Now(),
+		}
+		test.checkContexts[i] = checkContext
+
+		err = common.Retry(func() error {
+			if err = check(checkContext); err != nil {
+				return err
+			}
+			checkContext.successTime = time.Now()
+			if checkContext.time.IsZero() {
+				checkContext.time = checkContext.successTime
+			}
+			return nil
+		}, retries, time.Second)
+
+		if err != nil {
+			g := checkContext.getWholeGraph(t, checkContext.startTime)
+			f := checkContext.getAllFlows(t, checkContext.startTime)
+			helper.ExecCmds(t, test.tearDownCmds...)
+			context.getSystemState(t)
+			t.Errorf("Test failed: %s, graph: %s, flows: %s", err.Error(), g, f)
+			return
+		}
+	}
+
+	if test.tearDownFunction != nil {
+		if err = test.tearDownFunction(context); err != nil {
+			helper.ExecCmds(t, test.tearDownCmds...)
+			context.getSystemState(t)
+			t.Fatalf("Fail to tear test down: %s", err.Error())
+		}
+	}
+
+	helper.ExecCmds(t, test.tearDownCmds...)
+
+	if test.mode == Replay {
+		for i, check := range test.checks {
+			checkContext := test.checkContexts[i]
+			t.Logf("Replaying test with time %s (Unix: %d), startTime %s (Unix: %d)", checkContext.time, checkContext.time.Unix(), checkContext.startTime, checkContext.startTime.Unix())
+			err = common.Retry(func() error {
+				return check(checkContext)
+			}, retries, time.Second)
+
+			if err != nil {
+				t.Errorf("Failed to replay test: %s, graph: %s, flows: %s", err.Error(), checkContext.getWholeGraph(t, checkContext.time), checkContext.getAllFlows(t, checkContext.time))
+			}
+		}
+	}
+}
+
+func TestStartupCaptureNew(t *testing.T) {
+	test := &Test{
+		setupCmds: []helper.Cmd{
+			{"ovs-vsctl add-br br-startup", true},
```

Comment:
I think for this test you only need to create this bridge as you just want to test that the capture is started